llvm/lib/Target/AIE/AIEHazardRecognizer.h

```diff
@@ -178,12 +178,8 @@ class AIEHazardRecognizer : public ScheduleHazardRecognizer {
   unsigned computeScoreboardDepth() const;
 
 protected:
-  ScheduleHazardRecognizer::HazardType getHazardType(const MCInstrDesc &Desc,
-                                                     MemoryBankBits MemoryBanks,
-                                                     int DeltaCycles);
-  ScheduleHazardRecognizer::HazardType getHazardType(unsigned SchedClass,
-                                                     SlotBits SlotSet,
-                                                     MemoryBankBits MemoryBanks,
+  ScheduleHazardRecognizer::HazardType getHazardType(const MachineInstr *MI,
+                                                     const MCInstrDesc &Desc,
```

Comment:
Please only have `const MachineInstr &MI` and `int DeltaCycles` as parameters. We can easily retrieve `Desc` from `MI`.

Also note I changed the pointer to a reference.

Reply:
Oh I guess you do that because you want to allow a `Desc` that does not correspond to `MI.getDesc()`. I think we discussed that it would be nice not to bring a whole `MachineInstr` as parameter. We want to keep the scheduler open and flexible. In some cases, we won't have an MI to verify conflicts.

It seems you only need `MI` to compute the sched class. Could you then keep the old prototype and just add a `MIOperands` parameter? E.g.

```
ScheduleHazardRecognizer::HazardType getHazardType(
  const MCInstrDesc &Desc, MemoryBankBits MemoryBanks,
  iterator_range<const MachineOperand *> MIOperands, int DeltaCycles);
```

Reply:
As you mentioned i bring MI & Desc because I do not want to relate `Desc` & `MI.getDesc()`. Based on our last conversation regarding the reason of not bring `MachineInstr` was it would impact unit test, but that is not the case since we use `checkConflict` and `enterResources`.
 
With respect to possibility of not having `MI`, currently every where `MI` is available.
Have also added an interface as suggested which will work without `MI`. I believe we should have interface with MI because its very simple to use and will avoid possible mistake.

Reply:
The scheduler is evolving quite a bit, and in future cases, I can assume we won't have a full `MI` to pass around. This is also un-doing a previous decision of only passing what is necessary to `AIEHazardRecognizer` internal calls. In that regard, I believe we should only have the interface below, and get rid of any `MachineInstr`:
```
ScheduleHazardRecognizer::HazardType getHazardType(
  const MCInstrDesc &Desc, MemoryBankBits MemoryBanks,
  iterator_range<const MachineOperand *> MIOperands, int DeltaCycles);
```

Reply:
Have updated to one interface as suggested above.